jctools-core/src/main/java/org/jctools/queues/takestrategy/YieldTakeStrategy.java

```diff
@@ -0,0 +1,25 @@
+package org.jctools.queues.takestrategy;
+
+public class YieldTakeStrategy<E> implements TakeStrategy<E>
+{
+    @Override
+    public void signal()
+    {
+        // Nothing to do
+    }
+
+    @Override
+    public E waitFor(SupplierJDK6<E> supplier)
+    {
+        while(true) // Should introduce safepoints
```

Comment:
Thread.yield() should introduce the safepoint, no?
In any case this loop will be just as correct written naturally:

    E e;
    while((e = s.get()) == null)
      Thread.yield();
    return e;